src/linker/Linker.Steps/RemoveUnreachableBlocksStep.cs

```diff
@@ -194,8 +406,22 @@ bool TryInlineBodyDependencies (ref BodyReducer reducer)
 							break;
 					}
 
-					if (!TryGetConstantResultInstructionForMethod (md, out targetResult))
+					if (md == reducer.Body.Method) {
+						// Special case for direct recursion - simply assume non-const value
+						// since we can't tell.
 						break;
+					}
+
+					if (!TryGetConstantResultForMethod (md, out targetResult)) {
+						if (!treatUnprocessedDependenciesAsNonConst)
+							hasUnprocessedDependencies = true;
+						break;
+					} else if (targetResult == null || hasUnprocessedDependencies) {
+						// Even is const is detected, there's no point in rewriting anything
```

Comment:
I don't understand the comment - I thought that if `TryGetConstantResultForMethod` returns `true` the method is analyzed, so `targetResult == null` in this case means we've analyzed the method and know it is not constant. Why should we break in that case?

Reply:
It's non-const - so nothing to do - break in this case means continue with next instruction (break from the switch over instruction opcode).